### PR TITLE
HSEARCH-3985 Test Hibernate Search 6 indexing when hibernate.use_identifier_rollback=true

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-3985

### CODE REVIEW

File: integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/AutomaticIndexingIdentiferRollbackIT.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+			entity1Id.set( entity1.getId() );
+
+			backendMock.expectWorks( EntityWithJpaIdAsDocumentId.NAME )
+					.add( entity1.getId().toString(), b -> { } )
+					.processedThenExecuted();
+		} );
+		backendMock.verifyExpectationsMet();
+
+		AtomicReference<EntityWithJpaIdAsDocumentId> entity1FromTransaction = new AtomicReference<>();
```
Comment (fax4ever) [inline integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/AutomaticIndexingIdentiferRollbackIT.java:73]: it seems that this reference is set in the lambda but never get after that

**Review by fax4ever — COMMENTED** [review]
Looks good, I've just a comment here...

File: integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/AutomaticIndexingIdentiferRollbackIT.java (hunk 1 above)
Comment (yrodiere) [reply integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/AutomaticIndexingIdentiferRollbackIT.java]: Right, this was a remnant of a former attempt at testing this. It's not necessary, I removed it.

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**